lib/v2/screens/explore_screens/vote_screens/vote/interactor/viewmodels/vote_state.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:equatable/equatable.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/explore_screens/vote_screens/vote/interactor/viewmodels/current_remaining_time.dart';
+
+/// --- STATE
+class VoteState extends Equatable {
+  final PageState pageState;
+  final int remainingTimeStamp;
+  final CurrentRemainingTime? currentRemainingTime;
+  final String? errorMessage;
+
+  const VoteState({
+    required this.pageState,
+    required this.remainingTimeStamp,
+    this.currentRemainingTime,
+    this.errorMessage,
+  });
+
+  @override
+  List<Object?> get props => [
+        pageState,
+        remainingTimeStamp,
+        currentRemainingTime,
+        errorMessage,
+      ];
+
+  VoteState copyWith({
+    PageState? pageState,
+    int? remainingTimeStamp,
+    CurrentRemainingTime? currentRemainingTime,
+    String? errorMessage,
+  }) {
+    return VoteState(
+      pageState: pageState ?? this.pageState,
+      remainingTimeStamp: remainingTimeStamp ?? this.remainingTimeStamp,
+      currentRemainingTime: currentRemainingTime ?? this.currentRemainingTime,
+      errorMessage: errorMessage ?? this.errorMessage,
```

Comment:
How would you clear the error message? 
if you pass null to clear it, it will just use the old value?

Reply:
> it will just use the old value?

This is the real question.
When there is an error, the full error screen is usually shown and no more operations can be done so the errorMessage will never change, now we would have to rethink this if we change the full error screen.